src/backend/distributed/shared_library_init.c

```diff
@@ -1436,20 +1430,6 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
-	DefineCustomEnumVariable(
-		"citus.replication_model",
```

Comment:
I think we usually leave GUCs in place in "minor" versions to avoid unexpected errors/warnings, but change the description to be deprecated and make sure they have GUC_NO_SHOW_ALL.

Reply:
I added the GUC back with a warning message on SET queries. I realize that it is not covered in tests. I will add some tests to show (i) we see those warnings on SET queries (ii) changing the GUC has no affect on the behavior.